lib/immobilienscout/response_parser.rb

```diff
@@ -44,5 +44,11 @@ def id(response_body)
       messages = response_body['common.messages'].first['message']
       messages['id'] if messages.is_a? Hash
     end
+
+    def assign_response_body
+      JSON.parse(@response.body)
+    rescue JSON::ParserError
+      JSON.parse(Hash.from_xml(@response.body).to_json)
```

Comment:
What is the purpose of this JSON serialization/deserialization step?

Anyway, the response parser should also be a separate one for XML. Just because the JSON is invalid doesn't mean that the response contains valid XML.